src/lightning/app/cli/lightning_cli.py

```diff
@@ -76,11 +76,18 @@ def main() -> None:
     # Check environment and versions if not in the cloud and not testing
     is_testing = bool(int(os.getenv("LIGHTING_TESTING", "0")))
     if not is_testing and "LIGHTNING_APP_STATE_URL" not in os.environ:
-        # Enforce running in PATH Python
-        _check_environment_and_redirect()
-
-        # Check for newer versions and upgrade
-        _check_version_and_upgrade()
+        try:
+            # Enforce running in PATH Python
+            _check_environment_and_redirect()
+
+            # Check for newer versions and upgrade
+            _check_version_and_upgrade()
+        except SystemExit:
+            raise
+        except Exception:
+            # Note: We intentionally ignore all exceptions here so that we never panic if one of the above calls fails.
+            # If they fail for some reason users should still be able to continue with their command.
```

Comment:
This also means that we will never know if these checks break in the future. Is this acceptable?